core/templates.yml

```diff
@@ -594,6 +632,18 @@ strings:
     sequence_asset_version_name:            '@sequence_asset_publish_name.v{version}'
     shot_asset_version_name:                '@shot_asset_publish_name.v{version}'
 
+    # ASSET XGEN VERSION NAMES:
+    project_asset_version_name_xgen_file:        '@project_asset_publish_name.v{version}__[{palette_name}]'
+    sequence_asset_version_name_xgen_file:       '@sequence_asset_publish_name.v{version}__[{palette_name}]'
+    shot_asset_version_name_xgen_file:           '@shot_asset_publish_name.v{version}__[{palette_name}]'
```

Comment:
With the earlier approach of collecting xgen palettes as a separate item, The collector was collecting disconnected palettes as well because of the same path patterns. To resolve this issue I referred the collect (layer) render images example in which we are providing node names while collecting item. So followed the same thing here and made palette names optional and passing valid values during collection.

Apparently we moved this collection logics to the as session items properties but I didn't modify those already defined templates. But yeah, since we are passing entire dependency paths from the session, maybe no need of making those as optional fields. 

I will check that once and make the required updates. Thanks for pointing this out.

Reply:
Whatever the method of collection, this template looks wrong. making only palette_name optional as in this template means you are looking for a file with `name.v001__`. Only make something optional in a template if it is truly optional in real life i.e. if a path should be allowed without that part of the template.
Same goes for `xgen_collection_subdirs` as well - do we expect only xgen/collections to exist without a palette_name subfolder? If not, make it not optional.

Reply:
Ok, `palette_name` not an optional field in the updated templates and also confirmed that its working fine with items that are added as session item properties.